### PR TITLE
Port LiquidiyLayer PRs to v2

### PR DESCRIPTION
**Pull request by nambrot**
### Description

Ports over #1290 and #1482 to v2/main

### CODE REVIEW

File: typescript/infra/helm/liquidity-layer-relayers/templates/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "abacus.name" -}}
```
Comment (asaj) [inline typescript/infra/helm/liquidity-layer-relayers/templates/_helpers.tpl:4]: s/abacus/hyperlane

File: typescript/infra/helm/liquidity-layer-relayers/templates/circle-relayer-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    spec:
+      containers:
+      - name: circle-relayer
+        image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+        imagePullPolicy: IfNotPresent
+        command:
+        - ./node_modules/.bin/ts-node
+        - ./typescript/infra/scripts/middleware/circle-relayer.ts
+        - -e
+        - {{ .Values.abacus.runEnv }}
```
Comment (asaj) [inline typescript/infra/helm/liquidity-layer-relayers/templates/circle-relayer-deployment.yaml:23]: Ditto

File: typescript/infra/helm/liquidity-layer-relayers/templates/portal-relayer-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    spec:
+      containers:
+      - name: portal-relayer
+        image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+        imagePullPolicy: IfNotPresent
+        command:
+        - ./node_modules/.bin/ts-node
+        - ./typescript/infra/scripts/middleware/portal-relayer.ts
+        - -e
+        - {{ .Values.abacus.runEnv }}
```
Comment (asaj) [inline typescript/infra/helm/liquidity-layer-relayers/templates/portal-relayer-deployment.yaml:23]: ditto

File: typescript/infra/src/middleware/liquidity-layer-relayer.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    true,
+  );
+}
+
+function getLiquidityLayerRelayerHelmValues<Chain extends ChainName>(
+  agentConfig: AgentConfig<Chain>,
+  relayerConfig: LiquidityLayerRelayerConfig,
+) {
+  const values = {
+    abacus: {
```
Comment (asaj) [inline typescript/infra/src/middleware/liquidity-layer-relayer.ts:46]: ditto

**Review by asaj — COMMENTED** [review]
Minor comments, assuming this has more or less been reviewed already

File: solidity/contracts/middleware/liquidity-layer/adapters/PortalAdapter.sol [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+
+    /// @notice Hyperlane domain => Wormhole domain.
+    mapping(uint32 => uint16) public hyperlaneDomainToWormholeDomain;
+    /// @notice transferId => token address
+    mapping(bytes32 => address) public portalTransfersProcessed;
+
+    uint32 localDomain;
+    // We could technically use Portal's sequence number here but it doesn't
+    // get passed through, so we would have to parse the VAA twice
+    // 224 bits should be large enough and allows us to pack into a single slot
```
[the remark this answers is not in the document]
Comment (nambrot) [reply solidity/contracts/middleware/liquidity-layer/adapters/PortalAdapter.sol:28]: The comment here is for `nonce`.

The slot here is for the `transferId` which is `nonce//domain`. It's technically not stored but used as the key for the `portalTransfersProcessed` mapping

**Review by yorhodes — APPROVED** [review]

**Pull request merged**